src/winscard_svc.c

```diff
@@ -886,47 +886,52 @@ static LONG MSGRemoveContext(SCARDCONTEXT hContext, SCONTEXT * threadContext)
 		hCard = *(int32_t *)ptr;
 
 		/*
-		 * Unlock the sharing
+		 * Unlock the sharing. If the reader or handle already disappeared, skip
+		 * the disconnection part and just delete the orphan handle.
 		 */
 		rv = RFReaderInfoById(hCard, &rContext);
-		if (rv != SCARD_S_SUCCESS)
+		if (rv != SCARD_S_SUCCESS && rv != SCARD_E_READER_UNAVAILABLE &&
```

Comment:
Why do you test for `SCARD_E_READER_UNAVAILABLE`?
This value is never returned by `RFReaderInfoById()`

Reply:
`SCARD_E_READER_UNAVAILABLE` can be returned by `RFReaderInfoById()`: it's a corner case when the reader context is still there, but has a zero reference count.

See `RFReaderInfoById()` using the `REF_READER` macro, which calls `_RefReader()` (and returns its error code on error), which in turn has this code:
```
	if (0 == sReader->reference)
		return SCARD_E_READER_UNAVAILABLE;
```

Reply:
You are right. As always :-)
I got misled because `REF_READER()` is a macro. The `return` was hidden.

Reply:
No I just observed this error code in my test and initially got puzzled by it not less than you :-)